core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -140,6 +140,16 @@
         new Setting<>("node.mode", "network", Function.identity(), Property.NodeScope);
     public static final Setting<Boolean> NODE_INGEST_SETTING =
         Setting.boolSetting("node.ingest", true, Property.NodeScope);
+
+    /**
+     * controls whether the node is allowed to persist things like metadata to disk
+     * Note that this does not control whether the node stores actual indices (see
+     * {@link #NODE_DATA_SETTING}). However, if this is false, {@link #NODE_DATA_SETTING}
+     * and {@link #NODE_MASTER_SETTING} must also be false.
+     *
+     */
+    public static final Setting<Boolean> NODE_LOCAL_STORAGE_SETTING = Setting.boolSetting("node.local_storage", true, Property.NodeScope);
```

Comment:
In e57232f I've made it so the node id of the tribe client nodes is based on the node id of the parent and the tribe node name. I've still kept the `NODE_LOCAL_STORAGE_SETTING` flag so that the tribe client nodes don't persist this name on disk (as they share the same data path with their tribe parent node). The tribe parent node generates and persists its node id in the same way as any other kind of node.